### ISSUE TITLE
Quantum armor is somehow causing better questing to continually rescan inventory

### ISSUE THREAD

**Issue by bombcar** [GTNewHorizons/GT-New-Horizons-Modpack#9694]
### Your GTNH Discord Username

bombar#1042

### Your Pack Version

2.1.2.0 + serverside

### Your Server

mmcc

### Type of Server

Thermos

### Your Expectation

Don't spend 3-10% of tic time in better questing.

### The Reality

![image](https://user-images.githubusercontent.com/533931/152693081-47e903b6-2453-450c-9767-d4978e228bf1.png)

[warmroast.zip](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/8010729/warmroast.zip)

### Your Proposal

Make it so better questing won't check an inventory more than once a second even if it thinks it is changing, and/or investigate what could be causing quantum armor to continually update inventory.

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [ ] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

**Comment by bombcar** [comment]
there was another issue with large pipes during this, but the ic2/bq thing is something I've seen before.

**Comment by lordclips** [comment]
![image](https://user-images.githubusercontent.com/72330305/152694440-ad4be867-8a9d-43de-ac06-62142e34c70d.png)

IC2 does a weird update using a "toggleTimer" semi-often in the source for QuantumSuits, semi-often updating this timer if it exists and ic2.platform() is simulating in its source. This could be a potential root for the constant BQ checks as it thinks it's changing, though there's also a lot more going on with QuantumSuits that could, themselves, also be part of the issue. I contend that this seems to be the only one that acts unprompted, though, and it happens in 2 of the 3 switch cases.

**Comment by lordclips** [comment]
Another note: Should test other IC2 armors for similar behavior

**Comment by lordclips** [comment]
Having inspected both BQ and StandardQuesting further, it is looking like less of a BQ problem and more of one specifically with IC2, since the call that causes every subsequent BQ call actually starts in a Minecraft container class method call. I tried looking at the source but tbh I cant get shit to compile, but for BQ and StandardQuesting they arent doing anything but reacting to the updates sent from the game itself, and I can roughly speculate which method call it is.

Bottom line: IC2 armor ticks are inefficient specifically when noticed by the quest book, which is exactly the problem here, but it's not the quest book but IC2 causing it with its constant checks and NBT messiness

**Comment by bombcar** [comment]
@mitchej123 sez

> I'm 100% ok with making it not use any energy, and that'd indirectly solve this particular case
now... to figure out how to adjust that behavior
probably need to redirect `(ElectricItem.manager.use(itemStack, 1.0D, player)))` call from onArmorTick

Informally asking players on our server to turn off quantum night vision seems to have saved 2% on tic time.
